lang/java/avro/src/main/java/org/apache/avro/generic/GenericDatumWriter.java

```diff
@@ -174,13 +174,22 @@ protected void writeWithoutConversion(Schema schema, Object datum, Encoder out)
         error(schema, datum);
       }
     } catch (NullPointerException e) {
-      throw npe(e, " of " + schema.getFullName());
+      throw npeCceAte(e, " of " + schema.getFullName());
     }
   }
 
-  /** Helper method for adding a message to an NPE. */
-  protected NullPointerException npe(NullPointerException e, String s) {
-    NullPointerException result = new NullPointerException(e.getMessage() + s);
+  /** Helper method for adding a message to an NPE CCE ATE. */
+  protected <T extends Exception> T npeCceAte(T e, String s) {
```

Comment:
Is there any reason why the return value needs to be T?  If it were simply `RuntimeException`, you wouldn't need any of the casts below.

At the same time, why not decorate *any* runtime exception from writeField method (not just the three explicitly listed here)?

Super minor nit: the name `npeCceAte` is not particularly helpful!  I'd be tempted to use `createAnnotatedException()` or something explicit.

Reply:
Sorry for the slow response, thank you for your review, I use generics just to adapt to the three exceptions, but I neglected that they are all runtime exceptions, so I updated it.